enable all linters in CI; fix lint issues

jhump opened the pull request:
Most of the lint complaints were either the form (1) `if` block has a return statement so remove `else` block and outdent it (2) or function returns error value that is being ignored. For the latter, many of the cases were appropriate to ignore the error, so I just added `_ = ` to "consume" the error and get rid of the lint failures.

jhump commented on Makefile:
```diff
@@ -1,64 +1,56 @@
-SRCS := $(shell find . -name '*.go')
-PKGS := $(shell go list ./...)
```
These seemed a little odd since none of the tools actually needed them. This came from a contribution/pull request. Maybe their intent was to speed things up, to only scan the file system once ? Seems totally unnecessary, and removing bash loop from the calls below makes me like the lay of the land a little more.

jhump commented on Makefile:
```diff
@@ -1,64 +1,56 @@
-			exit 1; \
-		fi; \
-	done
+	golint -min_confidence 0.9 -set_exit_status ./...
```
No need for double-running `go lint` if we use `-set_exit_status`. (Unfortunately, `gofmt` has no such flag.)

Also, the default confidence level of 0.8 caused it to complain about an exported method name. I don't like the idea of changing names in exported API just to satisfy a linter, so I up'ed this to make the warning go-away. (Unfortunately, it doesn't seem like this tool has other knobs for making exceptions and/or white-listing violations)

jhump commented on testing/test_server.go:
```diff
@@ -204,10 +207,26 @@ func (TestServer) HalfDuplexCall(str grpc_testing.TestService_HalfDuplexCallServ
+	// be echoed back to the client as response headers. The format of the value
+	// is "key: val". To have the server reply with more than one response
+	// header, supply multiple values in request metadata.
+	MetadataReplyHeaders = "reply-with-headers"
```
The linter complained about exported symbols that had no Go doc, so I fleshed these out.

dragonsinth commented on cmd/grpcurl/grpcurl.go:
```diff
@@ -269,7 +269,7 @@ func main() {
 		}
 		if cc != nil {
-			cc.Close()
+			_ = cc.Close()
```
yick

jhump replied:
Yeah, I know.

dragonsinth commented on cmd/grpcurl/grpcurl.go:
```diff
@@ -235,7 +235,7 @@ func main() {
 			}
 			if *serverName != "" {
-				creds.OverrideServerName(*serverName)
+				_ = creds.OverrideServerName(*serverName)
```
warn?

jhump replied:
Doh! I was mechanically ignoring errors, but this is a real violation that needs to be fixed!

dragonsinth commented on grpcurl.go:
```diff
@@ -225,16 +225,16 @@ func ListMethods(source DescriptorSource, serviceName string) ([]string, error)
+		methods = append(methods, method.GetName())
+	}
+	sort.Strings(methods)
+	return methods, nil
```
In general, I kind of dislike a linter forcing you to do this; if you really don't need `sd` beyond the scope of the else statement.

dragonsinth commented on grpcurl.go:
```diff
@@ -345,9 +344,8 @@ func InvokeRpc(ctx context.Context, source DescriptorSource, cc *grpc.ClientConn
-	} else {
-		return invokeUnary(ctx, stub, mtd, handler, requestData, req)
 	}
+	return invokeUnary(ctx, stub, mtd, handler, requestData, req)
```
I just find this so useless and less readable :(

dragonsinth commented on grpcurl_test.go:
```diff
@@ -62,7 +66,9 @@ func TestMain(m *testing.M) {
 		grpc.WithInsecure(), grpc.WithBlock()); err != nil {
 		panic(err)
 	}
-	defer ccReflect.Close()
```
ok... this is where I draw the line, this is horrible.  Not better

dragonsinth commented on grpcurl_test.go:
```diff
@@ -87,7 +97,9 @@ func TestMain(m *testing.M) {
-	defer ccProtoset.Close()
+	defer func() {
+		_ = ccProtoset.Close()
+	}()
```
lk;ajdsflk;jsdafl;ksdajf

jhump commented:
@dragonsinth, I removed `golint` and `errcheck` from the tools run in CI. They are what produced the more egregious constructs that were in this PR. I reverted them and just kept the ones that I feel are improvements (or at worst, neutral).

PTAL
